packages/artifact/__tests__/upload-artifact.test.ts

```diff
@@ -351,4 +354,119 @@ describe('upload-artifact', () => {
 
     expect(uploadResp).rejects.toThrow()
   })
+
+  it('should throw an error uploading blob chunks get delayed', async () => {
+    const mockDate = new Date('2020-01-01')
+    const dirPath = path.join(__dirname, `plz-upload`)
+    if (!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, {recursive: true})
+    }
+    // eslint-disable-next-line @typescript-eslint/await-thenable
+    await fs.writeFile(
+      path.join(dirPath, 'file1.txt'),
+      'test file content',
+      err => {
+        if (err) {
+          throw err
+        }
+      }
+    )
+    // eslint-disable-next-line @typescript-eslint/await-thenable
+    await fs.writeFile(
+      path.join(dirPath, 'file2.txt'),
+      'test file content',
+      err => {
+        if (err) {
+          throw err
+        }
+      }
+    )
+    // eslint-disable-next-line @typescript-eslint/await-thenable
+    await fs.writeFile(
+      path.join(dirPath, 'file3.txt'),
+      'test file content',
+      err => {
+        if (err) {
+          throw err
+        }
+      }
+    )
```

Comment:
you can import the promise-wrapped version of fs's writeFile. these that you are awaiting aren't really doing what you are expecting, they are not returning a promise which is why you are tripping eslint

```js
import {writeFile} from 'fs/promises'
```

Reply:
you can also use `writeFileSync` from the normal fs package since this is only a test